astropy/logger.py

```diff
@@ -211,7 +211,7 @@ def _showwarning(self, *args, **kwargs):
                     mod_name = mod.__name__
                     break
         else:  # pragma: py2
-            for name, mod in sys.modules.items():
+            for name, mod in list(sys.modules.items()):
```

Comment:
I agree - maybe I can try and do a pull request to ``six`` to see if they'd accept those changes. In the mean time I can add a helper function for the logger specifically.